lib/matplotlib/axes/_axes.py

```diff
@@ -6638,22 +6657,40 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
                     "sets and %d colors were provided" % (nx, len(color)))
                 raise ValueError(error_message)
 
-        # If bins are not specified either explicitly or via range,
-        # we need to figure out the range required for all datasets,
-        # and supply that to np.histogram.
-        if not binsgiven and not input_empty:
+        hist_kwargs = dict()
+
+        # if the bin_range is not given, compute without nan numpy
+        # does not do this for us when guessing the range (but will
+        # happily ignore nans when computing the histogram).
+        if bin_range is None:
             xmin = np.inf
             xmax = -np.inf
             for xi in x:
-                if len(xi) > 0:
+                if len(xi):
                     xmin = min(xmin, np.nanmin(xi))
                     xmax = max(xmax, np.nanmax(xi))
-            bin_range = (xmin, xmax)
+            # make sure we have at least one non-nan value
+            # before we
+            if np.isfinite([xmin, xmax]).all():
+                bin_range = (xmin, xmax)
+
+        # If bins are not specified either explicitly or via range,
+        # we need to figure out the range required for all datasets,
+        # and supply that to np.histogram.
+        if not bins_array_given and not input_empty and len(x) > 1:
```

Comment:
the checks for input_empty and len(x) > 1 are redundant I think?

Reply:
The `len(x) > 1` is to avoid triggering the numerical issue in numpy described in https://github.com/matplotlib/matplotlib/pull/8638#issuecomment-467292466.

These are also testing slightly different things as `np.array([[], []])` fails the first one, but not the second.